### ISSUE TITLE
Empty check boxes need more contrast with adjacent colors

### ISSUE THREAD

**Issue by miguelsolorio** [microsoft/vscode#146834]
https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html

> For active controls any visual information provided that is necessary for a user to identify that a control is present and how to operate it must have **a minimum 3:1 contrast ratio with the adjacent colors**

> For designing focus indicators, selection indicators and user interface components that need to be perceived clearly, the following are examples that have sufficient contrast...
> Checkbox - empty | A black border on a white background indicates the checkbox.

All of our ~input controls~ checkboxes in the dark & light theme need to have a 3:1 contrast ratio between its background and adjacent color in the empty state. So either the border (if it has one) or the container's background.

|Dark|Light|
|--|--|
|<img width="191" alt="CleanShot 2022-04-05 at 09 57 59@2x" src="https://user-images.githubusercontent.com/35271042/161809200-34bf11c9-20a9-4bec-93c6-618b2d8a7a3c.png">|<img width="172" alt="CleanShot 2022-04-05 at 09 57 41@2x" src="https://user-images.githubusercontent.com/35271042/161809172-35b0ef8c-492f-40fd-b941-697f46717850.png">|
|<img width="280" alt="CleanShot 2022-04-05 at 09 32 32@2x" src="https://user-images.githubusercontent.com/35271042/161802463-3079f144-e197-4f60-99fd-4f43eb02d40c.png">|<img width="280" alt="CleanShot 2022-04-05 at 09 39 35@2x" src="https://user-images.githubusercontent.com/35271042/161803706-4b54955d-3b10-4ed6-9edf-dc319b9d8a79.png">|

cc @isidorn @roblourens @rzhao271

**Comment by chryw** [comment]
@misolori I think dropdown doesn't need 3:1 border if it won't be empty. This rule only applies to controls without text or icon inside. If it has text, then the text will be required to have 4.5:1 CCR from another rule (icon will have 3:1). Either of the content inside will satisfy the overall active indicator rule.

**Comment by Trandung5698** [comment]
Gửi từ điện thoại thông minh Samsung Galaxy của tôi.
-------- Tin nhắn gốc --------Từ: Cherry Wang ***@***.***> Ngày: 05/04/2022  23:58  (GMT+07:00) Đến: microsoft/vscode ***@***.***> Cc: Subscribed ***@***.***> Chủ đề: Re: [microsoft/vscode] Input controls need more contrast with
  adjacent colors (Issue #146834) 

@misolori I think dropdown doesn't need 3:1 border if it won't be empty. This rule only applies to controls without text or icon inside. If it has text, then the text will be required to have 4.5:1 CCR from another rule (icon will have 3:1). Either of the content inside will satisfy the overall active indicator rule.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/microsoft/vscode/issues/146834#issuecomment-1089034581",
"url": "https://github.com/microsoft/vscode/issues/146834#issuecomment-1089034581",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by rzhao271** [comment]
Verified with Light+ and Dark+ themes in the Settings editor.
The v2 versions of those themes might have an issue, though.
